src/ng-select/items-list.ts

```diff
@@ -199,7 +199,11 @@ export class ItemsList {
         if (this.lastSelectedItem && indexOfLastSelected > -1) {
             this._markedIndex = indexOfLastSelected;
         } else {
-            this._markedIndex = markDefault ? this.filteredItems.findIndex(x => !x.disabled) : -1;
+            if (!this._ngSelect.excludeGroupsFromDefaultSelection) {
```

Comment:
Invert logic

```ts
if (this._ngSelect.excludeGroupsFromDefaultSelection) {
   this._markedIndex = markDefault ? this.filteredItems.findIndex(x => !x.disabled && !x.children) : -1;
} else {
   this._markedIndex = markDefault ? this.filteredItems.findIndex(x => !x.disabled) : -1;
}
```

Reply:
Thank you for taking the time to review the PR. I have inverted the logic as suggested and fixed the conflicts.